lib/engine/route.rb

```diff
@@ -270,8 +255,8 @@ def find_connections(connections_a, connections_b, other_paths)
     private
 
     def restore_connections(connection_hexes)
-      possibilities = connection_hexes.map do |hexes|
-        hex_ids = hexes.map(&:id)
+      possibilities = connection_hexes.map do |hex_ids|
+        hexes = hex_ids.map { |hex_str| @game.hex_by_id(hex_str.split.first) }
```

Comment:
why did you change this to get the hexes here every time instead of on init?

Reply:
restore_connection is only called on init, so I'm not sure what you mean by "every time"
I moved the @game.hex_by_id lookup here from Action::run_routes. It shouldn't be called any more often.

I moved it here because I needed to leave connections_hexes in string format so that hex_ids was the same string value stored in the JSON so that the Connection::id generated for "ambiguous" intra-tile paths would match correctly.